ctru-rs/src/services/reference.rs

```diff
@@ -0,0 +1,44 @@
+use crate::Error;
+use std::sync::Mutex;
+pub(crate) struct ServiceReference {
+    counter: &'static Mutex<usize>,
+    close: Box<dyn Fn()>,
```

Comment:
I wonder if we should make this `dyn Fn() + Send + Sync`. I think this PR as-is would remove the `Send` and `Sync` auto-impls for the services that hold this struct.